pkg/daemon/ceph/osd/daemon.go

```diff
@@ -447,11 +447,6 @@ func getAvailableDevices(context *clusterd.Context, agent *OsdAgent) (*DeviceOsd
 		}
 		logger.Infof("device %q is available.", device.Name)
 
-		if device.Type == sys.PartType && agent.storeConfig.EncryptedDevice {
```

Comment:
Do we know why this was disabled in the past? Was there some fix from ceph that allows it now?

Reply:
Yes, [by this change](https://github.com/ceph/ceph/commit/159c30d29f009c07e51a4bd7cd5ecea8cf6c8739), Ceph supports raw mode encrypted OSD. This mode supports both disk and partition. On the other hand, lvm mode encrypted OSD, which Rook has used, only supports disk. So we disabled creating encrypted OSD on part.

In addition, this fix is in all Pacific, Quincy, and of course Reef. So it's safe to use raw mode without any conditions in master and v1.12.

Reply:
Thanks for the background, sounds good